vhdbuilder/packer/vhd-scanning-exe-on-vm.sh

```diff
@@ -23,7 +23,13 @@ elif [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "22.04" ]] && [[ "${ARCH
         echo "Azure CLI is not installed properly."
         exit 1
     fi
-elif [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "22.04" ]] && [[ "${ARCHITECTURE,,}" != "arm64" ]]; then
+elif [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "24.04" ]] && [[ "${ARCHITECTURE,,}" == "arm64" ]]; then
+    sudo apt-get install -y ca-certificates curl apt-transport-https lsb-release gnupg
+    curl -sL https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+    echo "deb [arch=arm64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
+    sudo apt-get update -y && sudo apt-get upgrade -y
+    sudo apt-get install -y azure-cli
```

Comment:
Just curious, is there a reason this is different from 22.04 arm64 (L17-25)? I think 2204 (jammy) also has arm64 az binary in this repo, why do we use pip for that?

Reply:
It can probably be refactored, there was a lot of testing / trial and error when figuring out how to create a VM and running the scanning scripts in order to generate the trivy report. It worked at the time, but I'll need to go back and look at if the pip code can be removed.